volatility3/framework/plugins/linux/module_extract.py

```diff
@@ -0,0 +1,85 @@
+# This file is Copyright 2025 Volatility Foundation and licensed under the Volatility Software License 1.0
+# which is available at https://www.volatilityfoundation.org/license/vsl-v1.0
+#
+import logging
+from typing import List
+
+import volatility3.framework.symbols.linux.utilities.module_extract as linux_utilities_module_extract
+from volatility3.framework import interfaces, renderers
+from volatility3.framework.configuration import requirements
+from volatility3.framework.renderers import format_hints
+from volatility3.framework.objects import utility
+
+vollog = logging.getLogger(__name__)
+
+
+class ModuleExtract(interfaces.plugins.PluginInterface):
+    """Recreates an ELF file from a specific address in the kernel"""
+
+    _required_framework_version = (2, 0, 0)
```

Comment:
And possibly this?  I've been a bit lax in updating the API file to say what was added when...  5:S

Reply:
I added ` framework.require_interface_version(*_required_framework_version)` in the utilities module_extract but not the plugins by accident. I can add it to the plugins, but not sure what else you wanted in extract. Is `2, 0, 0` the wrong value?

Reply:
I did a push for the plugin `require_interface_version` call

Reply:
Well, that's the very, very first version of the framework and I haven't checked your code to see if you use any features that were only introduced recently, so the safest thing to do would be to always rely on the latest framework version?  But if you use things like `tasks.get_thread` that apparently only appeared in `(2,1,0)`...

Reply:
I'd just set it to (2,25,0) unless you've tested it with older versions...

Reply:
(unfortunately, we stopped documenting it accurately at 2.5.0...  5:S)
https://github.com/volatilityfoundation/volatility3/blob/develop/API_CHANGES.md

Reply:
Ok I will bump to that. We should enforce the API_CHANGES updates starting now since this will be post-parity.

Reply:
Yeah, I'll see if I can backfill it over the weekend...

Reply:
Ok just pushed with both places at 25